src/App.tsx

```diff
@@ -1,27 +1,63 @@
 import React from 'react';
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  // eslint-disable-next-line
-  console.log(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+import { Clock } from './Components/Clock/Clock';
+
+type State = {
+  isClockVisible: boolean,
+  clockName: number,
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    isClockVisible: true,
+    clockName: 0,
+  };
+
+  visible = () => {
+    this.setState({ isClockVisible: true });
+  };
+
+  hidden = () => {
+    this.setState({ isClockVisible: false });
+  };
+
+  randomName = () => {
+    this.setState({ clockName: Math.round(Math.random() * 100) });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {this.state.isClockVisible && (
+          <p>
+            <Clock clockName={this.state.clockName} />
+          </p>
```

Comment:
Wrap Clock result to p instead of wrapping component 

```suggestion
          <Clock clockName={this.state.clockName} />
```